src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryAPI.groovy

```diff
@@ -0,0 +1,314 @@
+                    // failure
+                    String error = conn.getErrorStream().text
+                    LOGGER.log(Level.WARNING, "Failed to submit permissions target for ${name}: ${conn.responseCode} ${error}")
+                }
+            } catch (MalformedURLException ex) {
+                LOGGER.log(Level.WARNING, "Not a valid URL for ${payloadFile.name}", ex)
+            } catch (IOException ioe) {
+                LOGGER.log(Level.WARNING, "Failed sending PUT for ${payloadFile.name}", ioe)
+            }
+        }
+
+        /**
+         * Deletes the specified {@code name} using {@code apiUrl}.
+         * @param apiUrl the base URL to the deletion API
+         * @param name the name of the object to delete
+         * @param kind the human-readable kind of object being deleted (for a log message)
+         */
+        private static void delete(String apiUrl, String name, String kind) {
+            if (DRY_RUN_MODE) {
+                LOGGER.log(Level.INFO, "Dry-run mode: Skipping DELETE call for ${kind} ${name} at ${apiUrl}/${name}")
+                return
+            }
+            LOGGER.log(Level.INFO, "Deleting ${kind} ${name}")
+            try {
+                URL url = new URL(apiUrl + '/' + name)
+
+                HttpURLConnection conn = (HttpURLConnection) url.openConnection()
+                conn.setAuthenticator(AUTHENTICATOR)
+                conn.setRequestMethod('DELETE')
+                String response = conn.getInputStream().text
+                LOGGER.log(Level.INFO, response)
+            } catch (MalformedURLException ex) {
+                LOGGER.log(Level.WARNING, "Not a valid URL for ${name}", ex)
+            } catch (IOException ioe) {
+                LOGGER.log(Level.WARNING, "Failed sending DELETE for ${name}", ioe)
+            }
+        }
+
+        private static String toGeneratedName(String prefix, String name) {
+            name = prefix + name.replaceAll('[ /]', '_')
+            if (name.length() > 64) {
+                // Artifactory has an undocumented max length for permission target names of 64 chars (and possibly other types)
+                // If length is exceeded, use 55 chars of the prefix+name, separator, and 8 hopefully unique chars (prefix of name's SHA-256)
+                name = name.substring(0, 54) + '_' + sha256(name).substring(0, 7)
+            }
+            return name
+        }
+
+        private static String sha256(String str) {
+            LOGGER.log(Level.INFO, "Computing sha256 for string: " + str)
+            try {
+                MessageDigest digest = MessageDigest.getInstance("SHA-256")
+                digest.update(str.bytes)
+                return digest.digest().encodeHex().toString()
+            } catch (Exception e) {
+                LOGGER.log(Level.WARNING, "Failed to compute SHA-256 digest", e)
+                return '00000000000000000000000000000000'
```

Comment:
Srsly?

Reply:
🤷 SHA-256 is always there per platform spec.